### PR TITLE
[GE-630] Adjust deploy-storybook for Frontier usage

### PR DESCRIPTION
**Pull request by stefanteixeira**
[GE-630]

### Description

Extending `deploy-storybook` action to support deployments for frontier-pub project, from the Growth Squad. To optimize the process, I've added a couple of options to enable using an existing Storybook package or an existing Docker image, avoiding the need of building the package/image twice. Also, an option to be able to set a custom folder name on Jenkins, since all of our jobs are on a `Frontier` folder, which does not match the repository name.

Finally, fixing `set-output` warnings from `deploy-storybook` and `yarn-install` actions, since that [will be deprecated soon](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

- Changes were tested in the Jenkins jobs:
  - [Temploy](https://jenkins-build.toptal.net/job/Frontier/job/frontier-pub-storybook-temploy-helm-run/)
  - [Staging](https://jenkins.toptal.net/job/Frontier/job/frontier-pub-storybook-staging-deployment/)
- Applications available on [Temploy](https://frontier-pub-pr-4241-storybook-frontier-pub-s.toptal.rocks/) and [Staging](https://frontier-storybook-staging.toptal.net/)
- Related frontier-pub PR: https://github.com/toptal/frontier-pub/pull/4255
- Related inf-helm PR for Staging setup: https://github.com/toptal/inf-helm/pull/2060

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/davinci-github-actions/blob/master/_docs/contribution/changeset-guidelines.md) (if needed)

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

[GE-630]: https://toptal-core.atlassian.net/browse/GE-630?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: deploy-storybook/action.yml [hunk 1]
```diff
@@ -30,28 +30,50 @@ inputs:
...
 runs:
   using: composite
   steps:
     - name: Install Dependencies
+      if: ${{ inputs.use-prebuilt-package == 'false' && inputs.use-prebuilt-image == 'false' }}
       uses: toptal/davinci-github-actions/yarn-install@v4.4.2
 
     - name: Build Storybook
+      if: ${{ inputs.use-prebuilt-package == 'false' && inputs.use-prebuilt-image == 'false' }}
+      run: yarn "$BUILD_COMMAND"
```
Comment (dmaklygin) [inline deploy-storybook/action.yml:54]: 👍

File: deploy-storybook/action.yml [hunk 2]
```diff
@@ -30,28 +30,50 @@ inputs:
     description: Command to build Storybook with
     required: false
     default: storybook:build
+  use-prebuilt-package:
+    description: Whether if a prebuilt Storybook package should be used
```
Comment (separatio) [inline deploy-storybook/action.yml:34]:
```suggestion
    description: If a prebuilt Storybook package should be used
```

We should use either `if` or `whether`. Not both. :slightly_smiling_face: 
Also, `whether` should be used when you provide an alternative.
Example: `Whether a prebuilt Storybook package should be used or not`

File: deploy-storybook/action.yml [hunk 3]
```diff
@@ -30,28 +30,50 @@ inputs:
...
+    required: false
+    default: 'false'
+  use-prebuilt-image:
+    description: Whether if a prebuilt Storybook Docker image should be used
```
Comment (separatio) [inline deploy-storybook/action.yml:38]:
```suggestion
    description: If a prebuilt Storybook Docker image should be used
```

**Review by separatio — APPROVED** [review]
I had a couple of nits, but it looks good overall. Didn't test.

**Review by dmaklygin — APPROVED** [review]
Checked the code. Looks good. Seems, backward compatibility wasn't broken :) 
Good job!

File: deploy-storybook/action.yml [hunk 4]
```diff
@@ -30,28 +30,50 @@ inputs:
...
         BUILD_COMMAND: ${{ inputs.build-command }}
-      run: yarn "$BUILD_COMMAND"
 
     - name: Specify Repository and Release Name
       id: repo
       shell: bash
       run: |
-        echo ::set-output name=repository_name::${{ github.event.repository.name }}
-        echo ::set-output name=release_name::${{ github.event.repository.name }}-pr-${{ github.event.number || github.event.issue.number }}-storybook
+        folder_name=${{ inputs.jenkins-folder-name }}
```
Comment (OndrejTuma) [inline deploy-storybook/action.yml:63]: For security sakes, you should use it as env variable
```yaml
env:
  FOLDER_NAME: ${{ inputs.jenkins-folder-name }}
```

File: deploy-storybook/action.yml [hunk 5]
```diff
@@ -30,28 +30,50 @@ inputs:
...
 
     - name: Build Storybook
+      if: ${{ inputs.use-prebuilt-package == 'false' && inputs.use-prebuilt-image == 'false' }}
+      run: yarn "$BUILD_COMMAND"
       shell: bash
       env:
         BUILD_COMMAND: ${{ inputs.build-command }}
-      run: yarn "$BUILD_COMMAND"
 
     - name: Specify Repository and Release Name
```
Comment (OndrejTuma) [inline deploy-storybook/action.yml:59]:
```suggestion
    - name: Specify Repository, Jenkins Folder and Release Name
```

File: deploy-storybook/action.yml [hunk 6]
```diff
@@ -30,28 +30,50 @@ inputs:
...
       shell: bash
       run: |
-        echo ::set-output name=repository_name::${{ github.event.repository.name }}
-        echo ::set-output name=release_name::${{ github.event.repository.name }}-pr-${{ github.event.number || github.event.issue.number }}-storybook
+        folder_name=${{ inputs.jenkins-folder-name }}
+
+        if [[ "$folder_name" == '' ]]; then
+          echo folder_name=${{ github.event.repository.name }} >> $GITHUB_OUTPUT
+        else
+          echo folder_name=${{ inputs.jenkins-folder-name }} >> $GITHUB_OUTPUT
```
Comment (OndrejTuma) [inline deploy-storybook/action.yml:68]: and also use the env variable here 🙇 
```suggestion
          echo folder_name=$FOLDER_NAME >> $GITHUB_OUTPUT
```

File: deploy-storybook/action.yml (hunk 2 above)
Comment (stefanteixeira) [reply deploy-storybook/action.yml]: Fixed, thanks!

File: deploy-storybook/action.yml (hunk 4 above)
Comment (stefanteixeira) [reply deploy-storybook/action.yml]: Thanks! Fixed, as well as the other two comments below 👌

**Review by OndrejTuma — APPROVED** [review]
Looks great 👍 
Thanks for the improvement 🙇

**Review by OndrejTuma — CHANGES_REQUESTED** [review]
Oh, almost forgot - need to add a changeset:
run `yarn changeset` locally
or [click here](https://github.com/toptal/davinci-github-actions/new/use-deploy-storybook-action-frontier?filename=.changeset/pretty-llamas-sort.md&value=---%0A%22davinci-github-actions%22%3A%20patch%0A---%0A%0A%5BGE-630%5D%20Adjust%20deploy-storybook%20for%20Frontier%20usage%0A)

**Comment by stefanteixeira** [comment]
@OndrejTuma my bad! It's added now 👍

**Review by OndrejTuma — APPROVED** [review]
Perfect 👌 thank you very much

**Pull request merged**